File: JavaScript/selectionSort.js
/**
 * Function to perform Selection Sort on an array.
 * @param {number[]} arr - The array to be sorted.
 * @returns {number[]} - The sorted array.
 */
function selectionSort(arr) {
    let n = arr.length; // Get the length of the array

    // Outer loop to traverse through all array elements
    for (let i = 0; i < n - 1; i++) {
        let minIndex = i; // Assume the current index is the minimum

        // Inner loop to find the minimum element in the unsorted sublist
        for (let j = i + 1; j < n; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j; // Update minIndex if a smaller element is found
            }
        }

        // Swap the found minimum element with the first unsorted element
        if (minIndex !== i) {
            [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]]; // Swap using destructuring
        }
    }

    return arr; // Return the sorted array
}

// Example usage
const arrayToSort = [64, 34, 25, 12, 22, 11, 90];
const sortedArray = selectionSort(arrayToSort);
console.log("Sorted Array:", sortedArray);

File: Python/selectionSort.py
def selection_sort(arr):
    
    # Traverse through the entire array
    for i in range(len(arr)):
        # Assume the minimum is the first element in the unsorted section
        min_index = i
        
        # Check the element to find the minimum in the remaining unsorted section
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[min_index]:  # If a smaller element is found
                min_index = j  # Update the index of the minimum element
        
        # Swap the found minimum element with the first unsorted element
        arr[i], arr[min_index] = arr[min_index], arr[i]

    return arr  # Return the sorted array

# Example usage
if __name__ == "__main__":
    # Sample array to sort
    sample_array = [64, 25, 12, 22, 11]
    print("Original array:", sample_array)
    
    # Sorting the array
    sorted_array = selection_sort(sample_array)
    print("Sorted array:", sorted_array)
